bluepysnap/edges.py

```diff
@@ -27,8 +26,353 @@
+        """
+        if source is None and target is None:
+            raise BluepySnapError("Either `source` or `target` should be specified")
+
+        source_ids = self._circuit.nodes.ids(source)
+        target_ids = self._circuit.nodes.ids(target)
+
+        result = self._get_ids_from_pop(lambda x: (x.pathway_edges(source_ids, target_ids), x.name),
+                                        CircuitEdgeIds)
+
+        if properties:
+            result = self.properties(result, properties)
+        return result
+
+    def afferent_edges(self, node_id, properties=None):
+        """Get afferent edges for given ``node_id``.
+
+        Args:
+            node_id (int): Target node ID.
+            properties: An edge property name, a list of edge property names, or None.
+
+        Returns:
+            pandas.Series/pandas.DataFrame/list:
+                A pandas Series indexed by edge ID if ``properties`` is a string.
+                A pandas DataFrame indexed by edge ID if ``properties`` is a list.
+                A list of edge IDs, if ``properties`` is None.
+        """
+        return self.pathway_edges(source=None, target=node_id, properties=properties)
+
+    def efferent_edges(self, node_id, properties=None):
+        """Get efferent edges for given ``node_id``.
+
+        Args:
+            node_id: source node ID
+            properties: None / edge property name / list of edge property names
+
+        Returns:
+            List of edge IDs, if ``properties`` is None;
+            Pandas Series indexed by edge IDs if ``properties`` is string;
+            Pandas DataFrame indexed by edge IDs if ``properties`` is list.
+        """
+        return self.pathway_edges(source=node_id, target=None, properties=properties)
+
+    def pair_edges(self, source_node_id, target_node_id, properties=None):
+        """Get edges corresponding to ``source_node_id`` -> ``target_node_id`` connection.
+
+        Args:
+            source_node_id: source node ID
+            target_node_id: target node ID
+            properties: None / edge property name / list of edge property names
+
+        Returns:
+            List of edge IDs, if ``properties`` is None;
+            Pandas Series indexed by edge IDs if ``properties`` is string;
+            Pandas DataFrame indexed by edge IDs if ``properties`` is list.
+        """
+        return self.pathway_edges(
+            source=source_node_id, target=target_node_id, properties=properties
+        )
+
+    @staticmethod
+    def _add_circuit_ids(its, source, target):
+        """Generator comprehension adding the CircuitIds to the iterator.
+
+        Notes:
+            Using closures or lambda functions would result in override functions and so the
+            source and target would be the same for all the populations.
+        """
+        return ((CircuitNodeId(source, it[0]), CircuitNodeId(target, it[1]), it[2]) for it in its)
```

Comment:
ahahah, le mec en roue libre sur la generateur comprehension. noice

Maybe you could destructure `it` to have proper names instead of `it[0],it[1],it[2]`

Reply:
:sunglasses: 

Done for the splitting